File: navi/src/main/java/org/kexie/android/dng/navi/widget/NaviViewFragment.java
package org.kexie.android.dng.navi.widget;

import android.content.Context;
import android.os.Bundle;
import android.util.AttributeSet;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import com.amap.api.navi.AMapNaviView;

import java.util.Objects;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.fragment.app.Fragment;

public class NaviViewFragment extends Fragment
{
    private AMapNaviView interView;

    public AMapNaviView getInterView()
    {
        return Objects.requireNonNull(interView);
    }

    @Override
    public void onInflate(@NonNull Context context,
                          @NonNull AttributeSet attrs,
                          @Nullable Bundle savedInstanceState)
    {
        super.onInflate(context, attrs, savedInstanceState);
        if (interView == null)
        {
            interView = new AMapNaviView(context);
        }
    }

    @Nullable
    @Override
    public View onCreateView(@NonNull LayoutInflater inflater,
                             @Nullable ViewGroup container,
                             @Nullable Bundle savedInstanceState)
    {
        if (interView == null)
        {
            interView = new AMapNaviView(inflater.getContext());
        }
        return interView;
    }


    @Override
    public void onResume()
    {
        super.onResume();
        if (interView != null)
        {
            interView.onResume();
        }
    }

    @Override
    public void onSaveInstanceState(@NonNull Bundle outState)
    {
        super.onSaveInstanceState(outState);
        if (interView != null)
        {
            interView.onSaveInstanceState(outState);
        }
    }

    @Override
    public void onCreate(@Nullable Bundle savedInstanceState)
    {
        super.onCreate(savedInstanceState);
        if (interView != null)
        {
            interView.onCreate(savedInstanceState);
        }
    }

    @Override
    public void onDestroy()
    {
        super.onDestroy();
        if (interView != null)
        {
            interView.onDestroy();
            interView = null;
        }
    }

    @Override
    public void onPause()
    {
        super.onPause();
        if (interView != null)
        {
            interView.onPause();
        }
    }
}

File: navi/src/main/java/org/kexie/android/dng/navi/widget/NaviViewFragment.kt
package org.kexie.android.dng.navi.widget

import android.content.Context
import android.os.Bundle
import android.util.AttributeSet
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import androidx.fragment.app.Fragment
import com.amap.api.navi.AMapNaviView

typealias NaviView = AMapNaviView

class NaviViewFragment : Fragment() {
    var innerView: NaviView? = null
        private set

    override fun onCreateView(inflater: LayoutInflater,
                              container: ViewGroup?,
                              savedInstanceState: Bundle?): View? {
        return innerView
    }

    override fun onInflate(context: Context, attrs: AttributeSet, savedInstanceState: Bundle?) {
        super.onInflate(context, attrs, savedInstanceState)
        if (innerView == null) {
            innerView = NaviView(context)
        }
    }

    override fun onAttach(context: Context) {
        super.onAttach(context)
        if (innerView == null) {
            innerView = NaviView(context)
        }
    }

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        innerView!!.onCreate(savedInstanceState)
    }

    override fun onResume() {
        super.onResume()
        innerView!!.onResume()
    }

    override fun onPause() {
        super.onPause()
        innerView!!.onPause()
    }

    override fun onDestroy() {
        super.onDestroy()
        if (innerView != null) {
            innerView!!.onDestroy()
            innerView = null
        }
    }

    override fun onSaveInstanceState(outState: Bundle) {
        super.onSaveInstanceState(outState)
        innerView!!.onSaveInstanceState(outState)
    }
}
